s2spy/traintest.py

```diff
@@ -2,52 +2,94 @@
+            containing the labels 'train', 'test', and possibly 'skip'.
     """
-    cv = KFold(n_splits, shuffle=shuffle, **kwargs)
-    folds = cv.split(df)
+    assert isinstance(
+        data, (xr.Dataset, pd.DataFrame)
+    ), "Input data should be of type xr.Dataset or pd.DataFrame"
 
-    # Map folds to a new dataframe
-    output = pd.DataFrame(index=df.index)
-    for i, (train_index, test_index) in enumerate(folds):
-        col_name = f"fold_{i}"
-        output[col_name] = "skip"
-        output[col_name].iloc[train_index] = "train"
-        output[col_name].iloc[test_index] = "test"
+    if isinstance(data, xr.Dataset):
+        data = data.copy()  # otherwise the dataset is modified inplace
 
-    return output
+        assert (
+            data.anchor_year.size > 1
+        ), "Input data must have more than 1 anchor year to split."
 
+        folds = cv.split(data.anchor_year)
 
-def random_strat():
-    """random_strat cross-validator."""
-    raise NotImplementedError
+        fold_data = np.empty(
+            (cv.get_n_splits(data.anchor_year.size), data.anchor_year.size), dtype="<U6"
+        )
+
+        for i, (train_indices, test_indices) in enumerate(folds):
+            fold_data[i, :] = "skip"
+            fold_data[i, train_indices] = "train"
+            fold_data[i, test_indices] = "test"
+
+        data.expand_dims("fold")
+        data["fold"] = np.arange(fold_data.shape[0])
+        data["traintest"] = (["fold", "anchor_year"], fold_data)
+        data = data.set_coords("traintest")
+
+    else:
+        # split the anchor years
+        anchor_years = np.unique(data["anchor_year"])
+        assert (
+            anchor_years.size > 1
+        ), "Input data must have more than 1 anchor year to split."
+
+        folds = cv.split(anchor_years)
 
+        # label every row in the dataframe
+        for i, (train_indices, test_indices) in enumerate(folds):
+            # create column fold_# filled with "skip" in dataframe
+            col_name = f"fold_{i}"
+            data[col_name] = ""
 
-def timeseries_split():
-    """timeseries_split cross-validator."""
+            # create dictionary with anchor_years and train/test
+            indices = np.empty(anchor_years.size, dtype="<U6")
+            indices[:] = "skip"
+            indices[train_indices] = "train"
+            indices[test_indices] = "test"
+            train_test_dict = dict(zip(anchor_years, indices))
+
+            # map anchor_year row values with train/test
+            data[col_name] = data["anchor_year"].map(train_test_dict)
+
+    return data
+
+
+def leave_n_out():
+    """cross validator that leaves out n anchor years, instead of cv
+    defined on number of splits
+    """
+    # yet to be implemented
     raise NotImplementedError
 
 
-def iter_traintest(traintest_group, data, dim_function = None):
+def iter_traintest(traintest_group, data, dim_function=None):
```

Comment:
Instead of adding non-functional code to a repo that we'll implement "someday/maybe", it would be better to open an issue to discuss it. Shall we leave it out for now?